addons/website_sale_picking/models/delivery_carrier.py

```diff
@@ -11,11 +11,9 @@
 class DeliveryCarrier(models.Model):
     _inherit = 'delivery.carrier'
 
-    # Onsite delivery means the client comes to a physical store to get the products himself.
-    delivery_type = fields.Selection(selection_add=[
-        ('onsite', 'Pickup in store')
-    ], ondelete={'onsite': 'set default'})
-
+    delivery_type = fields.Selection(
+        selection_add=[('onsite', "In-store pickup")], ondelete={'onsite': 'set default'}
+    )
     warehouse_ids = fields.Many2many(string="Stores", comodel_name='stock.warehouse')
```

Comment:
Yes, already done, not yet pushed :)

Reply:
It's a known issue (that I find critical but my last feedback got thrashed and I didn't take the time to do it myself: https://www.odoo.com/odoo/all-tasks/2837453).

The current check_company checks doesn't allow company-restricted records on shared company record.
Since currently delivery carriers are shared between companies (It's dumb IMHO, but that's the current state of the code, unless you go on the carrier product and set a company on it) and warehouses are restricted to a specific company, you cannot specify a company on a carrier unless it's restricted to a specific company (with the current `check_company` logic).

You'll have to rely on a domain specified by hand for now.

Reply:
Oooooh, thank you!!